### PR TITLE
feat: create ops virtual module for each extension

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit adds a "virtual ops module" for each extension with ops.

All the enabled ops provided by the module can be imported like so:
```
import { op_name1, op_name2 } from "ext:<name>/ops";
```

Closes https://github.com/denoland/deno_core/issues/223

### CODE REVIEW

File: core/extensions.rs
Comment (bartlomieju) [inline core/extensions.rs:617]: Ideally we could remove this `true`, but I think we're gonna face many crashes before we migrate all the code to use the new system.

Comment (bartlomieju) [reply core/extensions.rs:617]: Also, it's unclear what would be interaction with enabled/disabled ops.

Comment (bartlomieju) [reply core/extensions.rs:617]: Eg. if I remove the `true` I get failures in `testing/unit/ops_buffer_test.ts` which also uses `ensureFastOps(true)`. But I can use `import { ... } from "ext:/...` there because it's a test file that will fail a check in module loader. 

Ideas?

Comment (mmastrac) [reply core/extensions.rs:617]: > Also, it's unclear what would be interaction with enabled/disabled ops.

I think we need to throw from disabled ops as there's no way to have an optional import is there?



> Eg. if I remove the `true` I get failures in `testing/unit/ops_buffer_test.ts` which also uses `ensureFastOps(true)`. But I can use `import { ... } from "ext:/...` there because it's a test file that will fail a check in module loader.
> 
> Ideas?

I think we'll just need to reexport here:

What about this?

```
import * as ops from "ext:...";
globalThis.testOps = ops;
```

**Comment by bartlomieju** [comment]
A few more things to consider:
1. What is we have an op (`op_foo`) that is defined in `foo` ext in `deno` repo, but then an embedder disables that op and provides another extension that defines that op (is that actually even possible?) - in such case the origin of the op is different and JS code that imports the op function would crash
2. Still not sure what to do with enabled/disabled ops

File: core/extensions.rs
Comment (bartlomieju) [reply core/extensions.rs:617]: This will become a non-issue once we land https://github.com/denoland/deno_core/pull/456

File: core/modules/loaders.rs
Comment (bartlomieju) [inline core/modules/loaders.rs:126]: Should be gone once we start using "new_synthetic_module"

**Comment by bartlomieju** [comment]
Discussed offline with @mmastrac and we're going to start with a single "vritual module" for all ops. We need to measure impact on having multiple modules on startup time and memory usage.

**Review by littledivy — COMMENTED** [review]
Should be ready for review. Summary of the changes I made:

- Store C-fn mapped fn pointer of the synthetic module eval function in external references table.
- Shortcut `get_stalled_top_level_await_message` when module is synthetic.
- Update internal module count assumptions from `2`.

File: core/modules/map.rs
Comment (bartlomieju) [inline core/modules/map.rs:250]: Revert

File: core/runtime/bindings.rs
Comment (bartlomieju) [inline core/runtime/bindings.rs:113]: Revert

File: core/runtime/bindings.rs
Comment (bartlomieju) [inline core/runtime/bindings.rs:298]: Revert

File: core/runtime/bindings.rs
Comment (bartlomieju) [inline core/runtime/bindings.rs:718]: Not correct

File: core/runtime/bindings.rs
Comment (bartlomieju) [inline core/runtime/bindings.rs:717]: Remove

File: core/runtime/jsruntime.rs
Comment (bartlomieju) [inline core/runtime/jsruntime.rs:1055]: Not needed

File: core/runtime/jsruntime.rs
Comment (bartlomieju) [inline core/runtime/jsruntime.rs:1038]: Most likely this conditional should be used

File: core/modules/module_map_data.rs
Comment (littledivy) [inline core/modules/module_map_data.rs:521]:
```suggestion
    assert_eq!(data.next_load_id as usize, modules.len());
```

Comment (bartlomieju) [reply core/modules/module_map_data.rs]: I'm not sure about that. I think we're leaking some detail here. I want to investigate this `next_load_id` before I merge this PR.

Comment (littledivy) [reply core/modules/module_map_data.rs]: IIUC `next_load_id` is only tracked for non internal modules.

File: core/modules/module_map_data.rs
Comment (mmastrac) [inline core/modules/module_map_data.rs:519]: Can we make this a `const`?

File: core/runtime/jsruntime.rs
Comment (mmastrac) [inline core/runtime/jsruntime.rs:1078]: FastString::StaticAscii?

File: core/runtime/tests/snapshot.rs
Comment (mmastrac) [inline core/runtime/tests/snapshot.rs:151]: `const`

File: testing/checkin/runtime/console.ts
Comment (mmastrac) [inline testing/checkin/runtime/console.ts:3]: ❤️

**Review by mmastrac — APPROVED** [review]
LGTM once build passes.

**Pull request merged**
